src/rust/src/rustzcash.rs

```diff
@@ -136,58 +136,48 @@ fn fixed_scalar_mult(from: &[u8; 32], p_g: &jubjub::SubgroupPoint) -> jubjub::Su
 /// be present even if there are no shielded components to verify.
 #[no_mangle]
 pub extern "C" fn librustzcash_init_zksnark_params(
-    #[cfg(not(target_os = "windows"))] spend_path: *const u8,
-    #[cfg(target_os = "windows")] spend_path: *const u16,
-    spend_path_len: usize,
-    #[cfg(not(target_os = "windows"))] output_path: *const u8,
-    #[cfg(target_os = "windows")] output_path: *const u16,
-    output_path_len: usize,
     #[cfg(not(target_os = "windows"))] sprout_path: *const u8,
     #[cfg(target_os = "windows")] sprout_path: *const u16,
     sprout_path_len: usize,
     load_proving_keys: bool,
 ) {
     PROOF_PARAMETERS_LOADED.call_once(|| {
         #[cfg(not(target_os = "windows"))]
-        let (spend_path, output_path, sprout_path) = {
-            (
-                OsStr::from_bytes(unsafe { slice::from_raw_parts(spend_path, spend_path_len) }),
-                OsStr::from_bytes(unsafe { slice::from_raw_parts(output_path, output_path_len) }),
-                if sprout_path.is_null() {
-                    None
-                } else {
-                    Some(OsStr::from_bytes(unsafe {
-                        slice::from_raw_parts(sprout_path, sprout_path_len)
-                    }))
-                },
-            )
+        let sprout_path = if sprout_path.is_null() {
+            None
+        } else {
+            Some(OsStr::from_bytes(unsafe {
+                slice::from_raw_parts(sprout_path, sprout_path_len)
+            }))
         };
 
         #[cfg(target_os = "windows")]
-        let (spend_path, output_path, sprout_path) = {
-            (
-                OsString::from_wide(unsafe { slice::from_raw_parts(spend_path, spend_path_len) }),
-                OsString::from_wide(unsafe { slice::from_raw_parts(output_path, output_path_len) }),
-                if sprout_path.is_null() {
-                    None
-                } else {
-                    Some(OsString::from_wide(unsafe {
-                        slice::from_raw_parts(sprout_path, sprout_path_len)
-                    }))
-                },
-            )
+        let sprout_path = if sprout_path.is_null() {
+            None
+        } else {
+            Some(OsString::from_wide(unsafe {
+                slice::from_raw_parts(sprout_path, sprout_path_len)
+            }))
         };
 
-        let (spend_path, output_path, sprout_path) = (
-            Path::new(&spend_path),
-            Path::new(&output_path),
-            sprout_path.as_ref().map(Path::new),
-        );
+        let sprout_path = sprout_path.as_ref().map(Path::new);
+
+        let sprout_vk = {
+            use bellman::groth16::{VerifyingKey, prepare_verifying_key};
+            let sprout_vk_bytes = include_bytes!("sprout-groth16.vk");
```

Comment:
I also need to check that this matches the verifying key from the old parameters.

Reply:
I'm resolving this because you've seemingly checked that now.